### PR TITLE
[AN-5024] Fix issue with profile picture.

### PR DESCRIPTION
**Pull request by JPonte**
Ticket: https://wearezeta.atlassian.net/browse/AN-5024
#### APK
[Download build #8621](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8621/artifact/build/artifact/wire-dev-PR711-8621.apk)
[Download build #8626](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8626/artifact/build/artifact/wire-dev-PR711-8626.apk)
[Download build #8627](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8627/artifact/build/artifact/wire-dev-PR711-8627.apk)
[Download build #8631](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8631/artifact/build/artifact/wire-dev-PR711-8631.apk)
[Download build #8633](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8633/artifact/build/artifact/wire-dev-PR711-8633.apk)
[Download build #8652](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8652/artifact/build/artifact/wire-dev-PR711-8652.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  def this(context: Context) = this(context, null)
+
+  implicit lazy val inj = context.getApplicationContext.asInstanceOf[WireApplication].injector
+  implicit lazy val ctx = context.getApplicationContext.asInstanceOf[WireApplication].eventContext
+
+  private val diameter: Int = getContext.getResources.getDimensionPixelSize(R.dimen.pref_account_icon_size)
+  private val icon: FilledCircularBackgroundDrawable = new FilledCircularBackgroundDrawable(Color.TRANSPARENT, diameter)
+  setIcon(DrawableUtils.drawableToBitmapDrawable(getContext.getResources, icon, diameter))
+
+  final private val selfModelObserver: ModelObserver[Self] = new ModelObserver[Self]() {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: Now the class is in Scala, we can use signals directly - inject a zms, get the `selfUserId` from the current `zms` instance, and then load a bitmap based on that user's id. Then the view doesn't even need to be set from the outside - we can delete `setSelfUser()`

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import com.waz.zclient.views.ImageController.WireImage
+import com.waz.zclient.{R, WireApplication}
+import net.xpece.android.support.preference.Preference
+
+class PicturePreference(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) extends Preference(context, attrs, defStyleAttr, defStyleRes) {
+  def this(context: Context, attrs: AttributeSet, defStyleAttr: Int) = this(context, attrs, defStyleAttr, R.style.Preference_Material)
+  def this(context: Context, attrs: AttributeSet) = this(context, attrs, R.attr.preferenceStyle)
+  def this(context: Context) = this(context, null)
+
+  implicit lazy val inj = context.getApplicationContext.asInstanceOf[WireApplication].injector
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: If we extend/use injector/injectable, we should get these for free - check out one of the other views (`ChatheadView`, for example)

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  implicit lazy val ctx = context.getApplicationContext.asInstanceOf[WireApplication].eventContext
+
+  private val diameter: Int = getContext.getResources.getDimensionPixelSize(R.dimen.pref_account_icon_size)
+  private val icon: FilledCircularBackgroundDrawable = new FilledCircularBackgroundDrawable(Color.TRANSPARENT, diameter)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: I know it was probably from copying the Java code across - but I find it much nicer if we get rid of these unnecessary `Type` declarations.

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  setIcon(DrawableUtils.drawableToBitmapDrawable(getContext.getResources, icon, diameter))
+
+  final private val selfModelObserver: ModelObserver[Self] = new ModelObserver[Self]() {
+    override def updated(model: Self): Unit = {
+
+      val wireImage = WireImage(AssetId(model.getPicture.getId))
+      val drawable: ImageAssetDrawable = new ImageAssetDrawable(Signal(wireImage), ImageAssetDrawable.ScaleType.CenterInside, ImageAssetDrawable.RequestBuilder.Round, Some(icon))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: Same here with the type declaration

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala (hunk 2 above)
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: I think that in the ChatheadView and other views they get it from the ViewHelper, but Preference is not a view.

Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/PicturePreference.scala]: Oh, right - might be good to have a `PreferenceHelper` then...

File: app/build.gradle [hunk 5]
```diff
@@ -91,6 +91,7 @@ android {
         dev {
             applicationId "com.waz.zclient.dev"
             versionName majorVersion + android.defaultConfig.versionCode + "-dev"
+            minSdkVersion 21
```
Comment (deanrobertcook) [inline app/build.gradle]: I'm still kinda checking out if this is a good solution or not... QA does testing using their qaDebug build, which will stop working if we don't fix the root cause, or they have to drop testing 4.4. Also, might be best to add a comment explaining what it's for, or just rebase when I make the commit for main...

**Comment by JPonte** [comment]
build again

**Pull request merged**